src/Stratis.Features.SQLiteWalletRepository/SQLiteWalletRepositoryExt.cs

```diff
@@ -0,0 +1,128 @@
+        {
+            var pubKeyScript = new Script(Encoders.Hex.DecodeData(address.PubKey)); // P2PK
+            PubKey pubKey = PayToPubkeyTemplate.Instance.ExtractScriptPubKeyParameters(pubKeyScript);
+            var scriptPubKey = new Script(Encoders.Hex.DecodeData(address.ScriptPubKey));
+
+            var res = new HdAddress()
+            {
+                Address = repo.ScriptAddressReader.GetAddressFromScriptPubKey(repo.Network, scriptPubKey),
+                Index = address.AddressIndex,
+                HdPath = repo.ToHdPath(address.AccountIndex, address.AddressType, address.AddressIndex),
+                ScriptPubKey = new Script(Encoders.Hex.DecodeData(address.ScriptPubKey)),
+                Pubkey = pubKeyScript
+            };
+
+            return res;
+        }
+
+        internal static TransactionData ToTransactionData(this SQLiteWalletRepository repo, HDTransactionData transactionData, IEnumerable<HDPayment> payments)
+        {
+            return new TransactionData()
+            {
+                Amount = new Money(transactionData.Value, MoneyUnit.BTC),
+                BlockHash = uint256.Parse(transactionData.OutputBlockHash),
+                BlockHeight = transactionData.OutputBlockHeight,
+                CreationTime = DateTimeOffset.FromUnixTimeSeconds(transactionData.OutputTxTime),
+                Id = uint256.Parse(transactionData.OutputTxId),
+                Index = transactionData.OutputIndex,
+                // These two are always updated and used in tandem so we update them from a single source value.
+                IsCoinBase = transactionData.OutputTxIsCoinBase == 1,
+                IsCoinStake = transactionData.OutputTxIsCoinBase == 1,
+                // IsPropagated  // Not used currently.
+                ScriptPubKey = new Script(Encoders.Hex.DecodeData(transactionData.RedeemScript)),
+                SpendingDetails = (transactionData.SpendTxId == null) ? null : new SpendingDetails()
+                {
+                    BlockHeight = transactionData.SpendBlockHeight,
+                    // BlockIndex // Not used currently.
+                    CreationTime = DateTimeOffset.FromUnixTimeSeconds((int)transactionData.SpendTxTime),
+                    IsCoinStake = transactionData.SpendTxIsCoinBase == 1,
+                    TransactionId = uint256.Parse(transactionData.SpendTxId),
+                    Payments = payments.Select(p => new PaymentDetails()
+                    {
+                         Amount = new Money((decimal)p.SpendValue, MoneyUnit.BTC),
+                         DestinationScriptPubKey = new Script(Encoders.Hex.DecodeData(p.SpendScriptPubKey)),
+                         OutputIndex = p.SpendIndex
+                    }).ToList()
+                }
+            };
+        }
+
+        internal static IEnumerable<Script> GetDestinations(this SQLiteWalletRepository repo, Script redeemScript)
```

Comment:
This is a scriptPubKey, not a redeemScript. Note the naming of method `GetTemplateFromScriptPubKey` too

Reply:
What constitutes a `ScriptPubKey` is either a bit volatile in its definition or not specific enough. I want to distinguish between the `ScriptPubKey` found on an address and the `ScriptPubKey` found on a `TxOut.ScriptPubKey`. Its not the same thing in the sense that they can't be compared directly.